dissect/target/helpers/fsutil.py

```diff
@@ -564,11 +571,10 @@ def open_decompress(
         # In a valid bz2 header the 4th byte is in the range b'1' ... b'9'.
         return bz2.open(file, mode, encoding=encoding, errors=errors, newline=newline)
 
-    if HAS_ZSTD and magic[:4] in [b"\xfd\x2f\xb5\x28", b"\x28\xb5\x2f\xfd"]:
-        # stream_reader is not seekable, so we have to resort to the less
-        # efficient decompressor which returns bytes.
-        return io.BytesIO(zstandard.decompress(file.read()))
-
+    if magic[:4] in [b"\xfd\x2f\xb5\x28", b"\x28\xb5\x2f\xfd"]:
+        if HAS_ZSTD:
+            return zstd.open(file, mode=mode, encoding=encoding, errors=errors, newline=newline)
+        raise UncompressionError("fail to uncompress zstd file, missing backport.zstd dependency")
```

Comment:
```suggestion
    if HAS_ZSTD and magic[:4] in [b"\xfd\x2f\xb5\x28", b"\x28\xb5\x2f\xfd"]:
        return zstd.open(file, mode=mode, encoding=encoding, errors=errors, newline=newline)

```
Maybe just keep this like this for the time being (similar to the other branches) and remove the separate error. I've made a separate issue to consider making this consistent with all other compression algorithms: https://github.com/fox-it/dissect.target/issues/1410